src/wiki/language.rs

```diff
@@ -1,65 +1,5 @@
 use serde::Serialize;
+use wiki_tui::parse_languages;
 
-macro_rules! language {
-    ($($lang:ident, $lang_name:expr, $lang_native:expr, $lang_code:expr);+) => {
-        #[derive(Debug, Clone, Serialize)]
-        pub enum Language {
-            $($lang,)*
-        }
-
-        impl Language {
-            /// Returns the data associated to the language. It's formatted like this:
-            /// (Language name in English, Local Language name, Language Code)
-            fn language_data(&self) -> (&str, &str, &str) {
-                match self {
-                    $(
-                    Language::$lang => (stringify!($lang), stringify!($lang_name), stringify!($lang_code)),
-                    )*
-                }
-            }
-
-            /// Returns the English name of the language
-            pub fn name(&self) -> &str {
-                self.language_data().0
-            }
-
-            /// Returns the local name of the language
-            pub fn local_name(&self) -> &str {
-                self.language_data().1
-            }
-
-            /// Returns the language code
-            pub fn code(&self) -> &str {
-                self.language_data().2
-            }
-        }
-
-        impl From<&str> for Language {
-            fn from(s: &str) -> Self {
-                match s.as_ref() {
-                    $(
-                    stringify!($lang_code) | stringify!($lang_name) => Language::$lang,
-                    )*
-                    _ => Language::default()
-                }
-            }
-        }
-
-        pub static LANGUAGES: &[Language] = &[$(Language::$lang,)*];
-    };
-}
-
-// format:
-// Identifier of the language, Language name in english, Local Language name, Language code
-language!(
-    Abkhaz, Abkhaz, аԥсшәа, ab;
-    Acehnese, Acehnese, Acèh, ace;
-    English, English, English, en;
-    German, German, Deutsch, de
-);
-
-impl Default for Language {
-    fn default() -> Self {
-        Language::English
-    }
-}
+// This macro parses all the available wikipedia languages into an enum
+parse_languages!("./languages.json");
```

Comment:
Because we could, in the future, add more data via this method, I think it would be best to put `languages.json` into a `data` folder. That way we don't pollute the root